### ISSUE TITLE
Add KATAKANA_OR_HIRAGANA script

### ISSUE THREAD

**Issue by kojiishi** [harfbuzz/harfbuzz#1473]
Blink currently collates Katakana to Hiragana because both will be mapped to `kana` in OpenType, and reducing the number of script runs, even only for this case, is surprisingly a big win for the performance of shaping Japanese text.

Adding `hrkt` script will help the situation clearer from the interface perspective. Appreciate your considerations.

**Comment by behdad** [comment]
Maybe @dscorbett can look into this?  We already added a private script for Zawgyi.  Is not too much work to do this.

**Comment by dscorbett** [comment]
What’s the issue? You like the behavior of 'kana' but you would prefer to call it 'hrkt' instead? I don’t see the point.

Or do you mean that code points with `\p{sc=Katakana_Or_Hiragana}` should be associated with the script tag 'kana', just as those with `\p{sc=Katakana}` and `\p{sc=Hiragana}` are? There are no such code points. They all have `\p{sc=Common}` now.

**Comment by behdad** [comment]
> What’s the issue? You like the behavior of 'kana' but you would prefer to call it 'hrkt' instead? I don’t see the point.

Since both scripts resolve to same OpenType script, suggestion is to add a new private tag for combined script so Chrome and some others can avoid breaking runs between those.

**Comment by behdad** [comment]
We don't even have a shaper for those.  So this sounds like just adding:
```
hb.hh:#define HB_SCRIPT_ KATAKANA_OR_HIRAGANA  ((hb_script_t) HB_TAG ('?','?','?','?'))
```
and make sure that converts to OT script.  For converting it back, I don't know what's best to do.

@kojiishi is `hrkt` what ICU uses for this?

**Comment by behdad** [comment]
https://unicode-org.github.io/icu-docs/apidoc/released/icu4j/com/ibm/icu/lang/UScript.html#KATAKANA_OR_HIRAGANA

**Comment by behdad** [comment]
ISO 15924 has:
Hrkt | 412 | Japanese syllabaries (alias for Hiragana + Katakana)
-- | -- | --

**Comment by twardoch** [comment]
So there is a collective ISO tag Hrkt plus the separate ISO tags Hira and Kana. Unicode maps to the separate tags. 

OpenType uses one tag, "kana" for both Hiragana & Katakana. 

So it sounds to me that HB should prefer the ISO "Hrkt" tag as it directly corresponds to the OT "kana" tag. 

ISO "Kana" is Katakana-only, and together with ISO "Hira", they should be handled in some way, well, differently. 

But I see the point being that the ISO "Hrkt" <> OT "kana" is the mapping that can go both ways, and using the collective "Hrkt" makes more sense for text shaping.

**Comment by twardoch** [comment]
And yes, there is some inconvenience coming from the fact that OT "kana" is the collective while ISO "Kana" is only Katakana, so OT "kana" maps best to ISO "Hrkt" (which is what I'd recommend doing of there is an OT > ISO mapping)

**Comment by twardoch** [comment]
I've opened https://github.com/fonttools/fonttools/issues/1892

**Comment by behdad** [comment]
Right.  But `hb_script_t` declares itself as Unicode scripts.  And we definitely don't want to diverge from that too much.

**Comment by khaledhosny** [comment]
Do we still want to do anything here?

**Comment by behdad** [comment]
I think this is still a good idea. Should be easy.
